### PR TITLE
Read commands, parse them and execute them.

### PR DESCRIPTION
**Pull request by radumarias**

### CODE REVIEW

File: src/cmd.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+/// Read command and args.
+pub fn read_cmd() -> (usize, String) {
+    let mut input = String::new();
+    let c = io::stdin().read_line(&mut input).unwrap();
+    (c, input)
+}
+
+/// Execute a command with args.
+pub fn exec_cmd(cmd: &str, args: Vec<String>) -> io::Result<(i32, String, String)> {
+    if cmd == "cd" {
```
Comment (cristiborta) [inline src/cmd.rs]: why not using enums?

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            std::process::exit(0)
+        }
+
+        // parse command and args
+        match parse::parse_cmd(input.as_str()) {
+            // parse ok
+            Ok((cmd, args)) => {
+                match cmd.as_str() {
+                    // process exit
+                    "exit" | "quit" | "bye" => std::process::exit(0),
```
Comment (cristiborta) [inline src/main.rs:24]: same as above

File: src/parse.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    };
+
+    // parse args
+    let args = iter.next();
+    return match args {
+        Some(c) => {
+            // we have args
+            match c.chars().next() {
+                // remove single quotes and double quotes
+                Some('\'') | Some('"') => {
```
Comment (cristiborta) [inline src/parse.rs]: why match for both `'` and `"` and then just treat case for last one only?

File: src/parse.rs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    let first_quote = c.chars().next().unwrap();
+                    let last_quote = c.chars().last().unwrap();
+                    // check for closing quote
+                    if first_quote != last_quote {
+                        return Err(io::Error::other("missing closing quote"));
+                    }
+                    let args = c.replace(first_quote, "");
```
Comment (cristiborta) [inline src/parse.rs]: What about `last_quote`

**Comment by cristiborta** [comment]
don't you also add some tests?

File: src/cmd.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
+use std::path::Path;
+use std::{env, io};
+
+/// Read command and args.
```
Comment (andreimarcut) [inline src/cmd.rs:8]: What is the meaning of the triple slash comment style?

Comment (radumarias) [reply src/cmd.rs:8]: javadoc like

File: src/cmd.rs (hunk 1 above)
Comment (radumarias) [reply src/cmd.rs]: using cost, as enum cannot have predef string and cannot use lib to extract variant name

**Pull request merged**